src/pip/_internal/utils/outdated.py

```diff
@@ -30,6 +32,13 @@
 logger = logging.getLogger(__name__)
 
 
+def _get_statefile_path(cache_dir, key):
```

Comment:
I would add a test of this function using specific, hard-coded values.

Reply:
Refactored to `_get_statefile_name` [here](https://github.com/pypa/pip/pull/6855/commits/ece6aacd9fb37ed94149053425f0c46d68a46055#diff-53b83fe57d94dba72021e5cbac099934R35-R39) and added tests [here](https://github.com/pypa/pip/pull/6855/commits/ece6aacd9fb37ed94149053425f0c46d68a46055#diff-d831e5898f8b0306f90468c8231853a8R138-R154).